In [4]:
import matplotlib.pyplot as plt
import pandas as pd
import time

df = pd.read_csv('Telco-Customer-Churn.csv')

### Table for all customers with Churn="Yes"

In [7]:
sub_df = df.loc[df['Churn']=='Yes']
sub_df

customerID  gender  SeniorCitizen Partner Dependents  tenure  \
2     3668-QPYBK    Male              0      No         No       2   
4     9237-HQITU  Female              0      No         No       2   
5     9305-CDSKC  Female              0      No         No       8   
8     7892-POOKP  Female              0     Yes         No      28   
13    0280-XJGEX    Male              0      No         No      49   
...          ...     ...            ...     ...        ...     ...   
7021  1699-HPSBG    Male              0      No         No      12   
7026  8775-CEBBJ  Female              0      No         No       9   
7032  6894-LFHLY    Male              1      No         No       1   
7034  0639-TSIQW  Female              0      No         No      67   
7041  8361-LTMKD    Male              1     Yes         No       4   

     PhoneService MultipleLines InternetService OnlineSecurity  ...  \
2             Yes            No             DSL            Yes  ...   
4             Yes            No     Fiber optic             No  ...   
5             Yes           Yes     Fiber optic             No  ...   
8             Yes           Yes     Fiber optic             No  ...   
13            Yes           Yes     Fiber optic             No  ...   
...           ...           ...             ...            ...  ...   
7021          Yes            No             DSL             No  ...   
7026          Yes            No             DSL             No  ...   
7032          Yes           Yes     Fiber optic             No  ...   
7034          Yes           Yes     Fiber optic            Yes  ...   
7041          Yes           Yes     Fiber optic             No  ...   

     DeviceProtection TechSupport StreamingTV StreamingMovies        Contract  \
2                  No          No          No              No  Month-to-month   
4                  No          No          No              No  Month-to-month   
5                 Yes          No         Yes             Yes  Month-to-month   
8                 Yes         Yes         Yes             Yes  Month-to-month   
13                Yes          No         Yes             Yes  Month-to-month   
...               ...         ...         ...             ...             ...   
7021               No         Yes         Yes              No        One year   
7026               No          No          No              No  Month-to-month   
7032               No          No          No              No  Month-to-month   
7034              Yes          No         Yes              No  Month-to-month   
7041               No          No          No              No  Month-to-month   

     PaperlessBilling              PaymentMethod MonthlyCharges  TotalCharges  \
2                 Yes               Mailed check          53.85        108.15   
4                 Yes           Electronic check          70.70        151.65   
5                 Yes           Electronic check          99.65         820.5   
8                 Yes           Electronic check         104.80       3046.05   
13                Yes  Bank transfer (automatic)         103.70        5036.3   
...               ...                        ...            ...           ...   
7021              Yes           Electronic check          59.80         727.8   
7026              Yes  Bank transfer (automatic)          44.20        403.35   
7032              Yes           Electronic check          75.75         75.75   
7034              Yes    Credit card (automatic)         102.95       6886.25   
7041              Yes               Mailed check          74.40         306.6   

     Churn  
2      Yes  
4      Yes  
5      Yes  
8      Yes  
13     Yes  
...    ...  
7021   Yes  
7026   Yes  
7032   Yes  
7034   Yes  
7041   Yes  

[1869 rows x 21 columns]

Out of the total 7044 customers Telco has data on, there are 1869 customers based on the table above that are no longer customers. Our next step, is to attempt to find any similiraties or reocurrences between the people who stop being customers and see if there are any trends. 

### Combination to find most propense people to leave Telco

Now, I will combine the following conditions to find which group of people is more likely to stop being a Telco customer: 

- SeniorCitizen
- Partner
- Dependents
- Tenure (of at least 12 months)
- PhoneService
- MultipleLines
- InternetService


### SeniorCitizen and Partner

In [22]:
df2=df.loc[(df['SeniorCitizen']== 1) & (df['Partner']== "Yes")]
df2

customerID  gender  SeniorCitizen Partner Dependents  tenure  \
30    3841-NFECX  Female              1     Yes         No      71   
31    4929-XIHVW    Male              1     Yes         No       2   
52    6575-SUVOI  Female              1     Yes         No      25   
53    7495-OOKFY  Female              1     Yes         No       8   
54    4667-QONEA  Female              1     Yes        Yes      60   
...          ...     ...            ...     ...        ...     ...   
7010  0723-DRCLG  Female              1     Yes         No       1   
7015  0666-UXTJO    Male              1     Yes         No      34   
7023  1035-IPQPU  Female              1     Yes         No      63   
7031  3605-JISKB    Male              1     Yes         No      55   
7041  8361-LTMKD    Male              1     Yes         No       4   

     PhoneService MultipleLines InternetService OnlineSecurity  ...  \
30            Yes           Yes     Fiber optic            Yes  ...   
31            Yes            No     Fiber optic             No  ...   
52            Yes           Yes             DSL            Yes  ...   
53            Yes           Yes     Fiber optic             No  ...   
54            Yes            No             DSL            Yes  ...   
...           ...           ...             ...            ...  ...   
7010          Yes           Yes     Fiber optic             No  ...   
7015          Yes            No     Fiber optic             No  ...   
7023          Yes           Yes     Fiber optic             No  ...   
7031          Yes           Yes             DSL            Yes  ...   
7041          Yes           Yes     Fiber optic             No  ...   

     DeviceProtection TechSupport StreamingTV StreamingMovies        Contract  \
30                Yes         Yes          No              No        Two year   
31                Yes          No         Yes             Yes  Month-to-month   
52                 No         Yes         Yes              No  Month-to-month   
53                 No          No          No              No  Month-to-month   
54                Yes         Yes          No             Yes        One year   
...               ...         ...         ...             ...             ...   
7010               No          No          No              No  Month-to-month   
7015              Yes          No         Yes              No  Month-to-month   
7023              Yes          No         Yes             Yes  Month-to-month   
7031               No          No          No              No        One year   
7041               No          No          No              No  Month-to-month   

     PaperlessBilling            PaymentMethod MonthlyCharges  TotalCharges  \
30                Yes  Credit card (automatic)          96.35       6766.95   
31                Yes  Credit card (automatic)          95.50        181.65   
52                Yes  Credit card (automatic)          69.50       1752.65   
53                Yes  Credit card (automatic)          80.65         633.3   
54                Yes  Credit card (automatic)          74.85       4456.35   
...               ...                      ...            ...           ...   
7010              Yes         Electronic check          74.45         74.45   
7015              Yes  Credit card (automatic)          85.20       2874.45   
7023              Yes         Electronic check         103.50        6479.4   
7031               No  Credit card (automatic)          60.00        3316.1   
7041              Yes             Mailed check          74.40         306.6   

     Churn  
30      No  
31      No  
52      No  
53     Yes  
54      No  
...    ...  
7010   Yes  
7015    No  
7023    No  
7031    No  
7041   Yes  

[573 rows x 21 columns]

In [23]:
df2=df.loc[(df['SeniorCitizen']== 0) & (df['Partner']== "Yes")]
df2

customerID  gender  SeniorCitizen Partner Dependents  tenure  \
0     7590-VHVEG  Female              0     Yes         No       1   
8     7892-POOKP  Female              0     Yes         No      28   
10    9763-GRSKD    Male              0     Yes        Yes      13   
12    8091-TTVAX    Male              0     Yes         No      58   
15    3655-SNQYZ  Female              0     Yes        Yes      69   
...          ...     ...            ...     ...        ...     ...   
7024  7398-LXGYX    Male              0     Yes         No      44   
7028  9281-CEDRU  Female              0     Yes         No      68   
7038  6840-RESVB    Male              0     Yes        Yes      24   
7039  2234-XADUH  Female              0     Yes        Yes      72   
7040  4801-JZAZL  Female              0     Yes        Yes      11   

     PhoneService     MultipleLines InternetService OnlineSecurity  ...  \
0              No  No phone service             DSL             No  ...   
8             Yes               Yes     Fiber optic             No  ...   
10            Yes                No             DSL            Yes  ...   
12            Yes               Yes     Fiber optic             No  ...   
15            Yes               Yes     Fiber optic            Yes  ...   
...           ...               ...             ...            ...  ...   
7024          Yes               Yes     Fiber optic            Yes  ...   
7028          Yes                No             DSL             No  ...   
7038          Yes               Yes             DSL            Yes  ...   
7039          Yes               Yes     Fiber optic             No  ...   
7040           No  No phone service             DSL            Yes  ...   

     DeviceProtection TechSupport StreamingTV StreamingMovies        Contract  \
0                  No          No          No              No  Month-to-month   
8                 Yes         Yes         Yes             Yes  Month-to-month   
10                 No          No          No              No  Month-to-month   
12                Yes          No         Yes             Yes        One year   
15                Yes         Yes         Yes             Yes        Two year   
...               ...         ...         ...             ...             ...   
7024              Yes          No          No              No  Month-to-month   
7028               No         Yes         Yes              No        Two year   
7038              Yes         Yes         Yes             Yes        One year   
7039              Yes          No         Yes             Yes        One year   
7040               No          No          No              No  Month-to-month   

     PaperlessBilling              PaymentMethod MonthlyCharges  TotalCharges  \
0                 Yes           Electronic check          29.85         29.85   
8                 Yes           Electronic check         104.80       3046.05   
10                Yes               Mailed check          49.95        587.45   
12                 No    Credit card (automatic)         100.35        5681.1   
15                 No    Credit card (automatic)         113.25       7895.15   
...               ...                        ...            ...           ...   
7024              Yes    Credit card (automatic)          84.80       3626.35   
7028               No  Bank transfer (automatic)          64.10       4326.25   
7038              Yes               Mailed check          84.80        1990.5   
7039              Yes    Credit card (automatic)         103.20        7362.9   
7040              Yes           Electronic check          29.60        346.45   

     Churn  
0       No  
8      Yes  
10      No  
12      No  
15      No  
...    ...  
7024    No  
7028    No  
7038    No  
7039    No  
7040    No  

[2829 rows x 21 columns]

In [24]:
df2=df.loc[(df['SeniorCitizen']== 1) & (df['Partner']== "No")]
df2

customerID  gender  SeniorCitizen Partner Dependents  tenure  \
20    8779-QRDMV    Male              1      No         No       1   
34    3413-BMNZE    Male              1      No         No       1   
50    8012-SOUDQ  Female              1      No         No      43   
55    1658-BYGOY    Male              1      No         No      18   
75    2673-CXQEU  Female              1      No         No      56   
...          ...     ...            ...     ...        ...     ...   
6980  0786-VSSUD    Male              1      No         No      16   
6987  5981-ZVXOT  Female              1      No         No      35   
7008  1980-KXVPM  Female              1      No         No       3   
7029  2235-DWLJU  Female              1      No         No       6   
7032  6894-LFHLY    Male              1      No         No       1   

     PhoneService     MultipleLines InternetService OnlineSecurity  ...  \
20             No  No phone service             DSL             No  ...   
34            Yes                No             DSL             No  ...   
50            Yes               Yes     Fiber optic             No  ...   
55            Yes               Yes     Fiber optic             No  ...   
75            Yes               Yes     Fiber optic            Yes  ...   
...           ...               ...             ...            ...  ...   
6980           No  No phone service             DSL             No  ...   
6987          Yes               Yes     Fiber optic            Yes  ...   
7008          Yes               Yes     Fiber optic             No  ...   
7029           No  No phone service             DSL             No  ...   
7032          Yes               Yes     Fiber optic             No  ...   

     DeviceProtection TechSupport StreamingTV StreamingMovies        Contract  \
20                Yes          No          No             Yes  Month-to-month   
34                 No          No          No              No  Month-to-month   
50                 No          No         Yes              No  Month-to-month   
55                 No          No         Yes             Yes  Month-to-month   
75                Yes          No         Yes             Yes        One year   
...               ...         ...         ...             ...             ...   
6980               No          No         Yes              No  Month-to-month   
6987              Yes          No         Yes             Yes        One year   
7008               No          No          No              No  Month-to-month   
7029               No          No         Yes             Yes  Month-to-month   
7032               No          No          No              No  Month-to-month   

     PaperlessBilling              PaymentMethod MonthlyCharges  TotalCharges  \
20                Yes           Electronic check          39.65         39.65   
34                 No  Bank transfer (automatic)          45.25         45.25   
50                Yes           Electronic check          90.25       3838.75   
55                Yes           Electronic check          95.45       1752.55   
75                 No           Electronic check         110.50        6139.5   
...               ...                        ...            ...           ...   
6980              Yes               Mailed check          39.00        679.85   
6987              Yes           Electronic check         108.35       3726.15   
7008              Yes    Credit card (automatic)          75.05        256.25   
7029              Yes           Electronic check          44.40        263.05   
7032              Yes           Electronic check          75.75         75.75   

     Churn  
20     Yes  
34      No  
50      No  
55     Yes  
75      No  
...    ...  
6980   Yes  
6987    No  
7008   Yes  
7029    No  
7032   Yes  

[569 rows x 21 columns]

In [25]:
df2=df.loc[(df['SeniorCitizen']== 0) & (df['Partner']== "No")]
df2

customerID  gender  SeniorCitizen Partner Dependents  tenure  \
1     5575-GNVDE    Male              0      No         No      34   
2     3668-QPYBK    Male              0      No         No       2   
3     7795-CFOCW    Male              0      No         No      45   
4     9237-HQITU  Female              0      No         No       2   
5     9305-CDSKC  Female              0      No         No       8   
...          ...     ...            ...     ...        ...     ...   
7034  0639-TSIQW  Female              0      No         No      67   
7035  8456-QDAVC    Male              0      No         No      19   
7036  7750-EYXWZ  Female              0      No         No      12   
7037  2569-WGERO  Female              0      No         No      72   
7042  3186-AJIEK    Male              0      No         No      66   

     PhoneService     MultipleLines InternetService       OnlineSecurity  ...  \
1             Yes                No             DSL                  Yes  ...   
2             Yes                No             DSL                  Yes  ...   
3              No  No phone service             DSL                  Yes  ...   
4             Yes                No     Fiber optic                   No  ...   
5             Yes               Yes     Fiber optic                   No  ...   
...           ...               ...             ...                  ...  ...   
7034          Yes               Yes     Fiber optic                  Yes  ...   
7035          Yes                No     Fiber optic                   No  ...   
7036           No  No phone service             DSL                   No  ...   
7037          Yes                No              No  No internet service  ...   
7042          Yes                No     Fiber optic                  Yes  ...   

         DeviceProtection          TechSupport          StreamingTV  \
1                     Yes                   No                   No   
2                      No                   No                   No   
3                     Yes                  Yes                   No   
4                      No                   No                   No   
5                     Yes                   No                  Yes   
...                   ...                  ...                  ...   
7034                  Yes                   No                  Yes   
7035                   No                   No                  Yes   
7036                  Yes                  Yes                  Yes   
7037  No internet service  No internet service  No internet service   
7042                  Yes                  Yes                  Yes   

          StreamingMovies        Contract PaperlessBilling  \
1                      No        One year               No   
2                      No  Month-to-month              Yes   
3                      No        One year               No   
4                      No  Month-to-month              Yes   
5                     Yes  Month-to-month              Yes   
...                   ...             ...              ...   
7034                   No  Month-to-month              Yes   
7035                   No  Month-to-month              Yes   
7036                  Yes        One year               No   
7037  No internet service        Two year              Yes   
7042                  Yes        Two year              Yes   

                  PaymentMethod MonthlyCharges  TotalCharges Churn  
1                  Mailed check          56.95        1889.5    No  
2                  Mailed check          53.85        108.15   Yes  
3     Bank transfer (automatic)          42.30       1840.75    No  
4              Electronic check          70.70        151.65   Yes  
5              Electronic check          99.65         820.5   Yes  
...                         ...            ...           ...   ...  
7034    Credit card (automatic)         102.95       6886.25   Yes  
7035  Bank transfer (automatic)          7

### SeniorCitizen and Dependents

In [26]:
df2=df.loc[(df['SeniorCitizen']== 0) & (df['Dependents']== "Yes")]
df2

customerID  gender  SeniorCitizen Partner Dependents  tenure  \
6     1452-KIOVK    Male              0      No        Yes      22   
9     6388-TABGU    Male              0      No        Yes      62   
10    9763-GRSKD    Male              0     Yes        Yes      13   
15    3655-SNQYZ  Female              0     Yes        Yes      69   
17    9959-WOFKT    Male              0      No        Yes      71   
...          ...     ...            ...     ...        ...     ...   
7018  1122-JWTJW    Male              0     Yes        Yes       1   
7020  9837-FWLCH    Male              0     Yes        Yes      12   
7038  6840-RESVB    Male              0     Yes        Yes      24   
7039  2234-XADUH  Female              0     Yes        Yes      72   
7040  4801-JZAZL  Female              0     Yes        Yes      11   

     PhoneService     MultipleLines InternetService       OnlineSecurity  ...  \
6             Yes               Yes     Fiber optic                   No  ...   
9             Yes                No             DSL                  Yes  ...   
10            Yes                No             DSL                  Yes  ...   
15            Yes               Yes     Fiber optic                  Yes  ...   
17            Yes               Yes     Fiber optic                  Yes  ...   
...           ...               ...             ...                  ...  ...   
7018          Yes                No     Fiber optic                   No  ...   
7020          Yes                No              No  No internet service  ...   
7038          Yes               Yes             DSL                  Yes  ...   
7039          Yes               Yes     Fiber optic                   No  ...   
7040           No  No phone service             DSL                  Yes  ...   

         DeviceProtection          TechSupport          StreamingTV  \
6                      No                   No                  Yes   
9                      No                   No                   No   
10                     No                   No                   No   
15                    Yes                  Yes                  Yes   
17                    Yes                   No                  Yes   
...                   ...                  ...                  ...   
7018                   No                   No                   No   
7020  No internet service  No internet service  No internet service   
7038                  Yes                  Yes                  Yes   
7039                  Yes                   No                  Yes   
7040                   No                   No                   No   

          StreamingMovies        Contract PaperlessBilling  \
6                      No  Month-to-month              Yes   
9                      No        One year               No   
10                     No  Month-to-month              Yes   
15                    Yes        Two year               No   
17                    Yes        Two year               No   
...                   ...             ...              ...   
7018                   No  Month-to-month              Yes   
7020  No internet service  Month-to-month              Yes   
7038                  Yes        One year              Yes   
7039                  Yes        One year              Yes   
7040                   No  Month-to-month              Yes   

                  PaymentMethod MonthlyCharges  TotalCharges Churn  
6       Credit card (automatic)          89.10        1949.4    No  
9     Bank transfer (automatic)          56.15       3487.95    No  
10                 Mailed check          49.95        587.45    No  
15      Credit card (automatic)         113.25       7895.15    No  
17    Bank transfer (automatic)         106.70       7382.25    No  
...                         ...            ...           ...   ...  
7018               Mailed check          70.65         70.65   Yes  
7020           Electronic check          1

In [27]:
df2=df.loc[(df['SeniorCitizen']== 1) & (df['Dependents']== "Yes")]
df2

customerID  gender  SeniorCitizen Partner Dependents  tenure  \
54    4667-QONEA  Female              1     Yes        Yes      60   
57    5067-XJQFU    Male              1     Yes        Yes      66   
72    1891-QRQSA    Male              1     Yes        Yes      64   
140   3146-MSEGF  Female              1     Yes        Yes      72   
308   3167-SNQPL    Male              1     Yes        Yes      38   
...          ...     ...            ...     ...        ...     ...   
6235  5180-UCIIQ    Male              1     Yes        Yes      22   
6346  9661-MHUMO    Male              1     Yes        Yes       1   
6446  2205-LPVGL    Male              1     Yes        Yes      58   
6451  4868-AADLV    Male              1     Yes        Yes      66   
6763  7471-MQPOS    Male              1     Yes        Yes      71   

     PhoneService     MultipleLines InternetService OnlineSecurity  ...  \
54            Yes                No             DSL            Yes  ...   
57            Yes               Yes     Fiber optic             No  ...   
72            Yes               Yes     Fiber optic            Yes  ...   
140           Yes               Yes             DSL            Yes  ...   
308           Yes               Yes     Fiber optic             No  ...   
...           ...               ...             ...            ...  ...   
6235           No  No phone service             DSL            Yes  ...   
6346          Yes                No             DSL             No  ...   
6446          Yes               Yes     Fiber optic            Yes  ...   
6451          Yes               Yes     Fiber optic            Yes  ...   
6763          Yes               Yes             DSL             No  ...   

     DeviceProtection TechSupport StreamingTV StreamingMovies        Contract  \
54                Yes         Yes          No             Yes        One year   
57                Yes         Yes         Yes             Yes        One year   
72                Yes         Yes         Yes             Yes        Two year   
140               Yes         Yes         Yes             Yes        Two year   
308                No          No         Yes             Yes  Month-to-month   
...               ...         ...         ...             ...             ...   
6235               No          No          No             Yes  Month-to-month   
6346               No         Yes          No              No  Month-to-month   
6446               No         Yes          No              No        One year   
6451              Yes         Yes         Yes             Yes        One year   
6763              Yes         Yes          No             Yes        One year   

     PaperlessBilling              PaymentMethod MonthlyCharges  TotalCharges  \
54                Yes    Credit card (automatic)          74.85       4456.35   
57                Yes           Electronic check         108.45       7076.35   
72                Yes  Bank transfer (automatic)         111.60          7099   
140               Yes    Credit card (automatic)          88.05       6425.65   
308                No           Electronic check         101.15       3741.85   
...               ...                        ...            ...           ...   
6235               No               Mailed check          40.05         880.2   
6346              Yes    Credit card (automatic)          49.25         49.25   
6446              Yes  Bank transfer (automatic)          89.85       5125.75   
6451              Yes           Electronic check         116.25       7862.25   
6763              Yes  Bank transfer (automatic)          72.90       5139.65   

     Churn  
54      No  
57      No  
72      No  
140     No  
308     No  
...    ...  
6235   Yes  
6346   Yes  
6446    No  
6451    No  
6763    No  

[91 rows x 21 columns]

In [28]:
df2=df.loc[(df['SeniorCitizen']== 0) & (df['Dependents']== "No")]
df2

customerID  gender  SeniorCitizen Partner Dependents  tenure  \
0     7590-VHVEG  Female              0     Yes         No       1   
1     5575-GNVDE    Male              0      No         No      34   
2     3668-QPYBK    Male              0      No         No       2   
3     7795-CFOCW    Male              0      No         No      45   
4     9237-HQITU  Female              0      No         No       2   
...          ...     ...            ...     ...        ...     ...   
7034  0639-TSIQW  Female              0      No         No      67   
7035  8456-QDAVC    Male              0      No         No      19   
7036  7750-EYXWZ  Female              0      No         No      12   
7037  2569-WGERO  Female              0      No         No      72   
7042  3186-AJIEK    Male              0      No         No      66   

     PhoneService     MultipleLines InternetService       OnlineSecurity  ...  \
0              No  No phone service             DSL                   No  ...   
1             Yes                No             DSL                  Yes  ...   
2             Yes                No             DSL                  Yes  ...   
3              No  No phone service             DSL                  Yes  ...   
4             Yes                No     Fiber optic                   No  ...   
...           ...               ...             ...                  ...  ...   
7034          Yes               Yes     Fiber optic                  Yes  ...   
7035          Yes                No     Fiber optic                   No  ...   
7036           No  No phone service             DSL                   No  ...   
7037          Yes                No              No  No internet service  ...   
7042          Yes                No     Fiber optic                  Yes  ...   

         DeviceProtection          TechSupport          StreamingTV  \
0                      No                   No                   No   
1                     Yes                   No                   No   
2                      No                   No                   No   
3                     Yes                  Yes                   No   
4                      No                   No                   No   
...                   ...                  ...                  ...   
7034                  Yes                   No                  Yes   
7035                   No                   No                  Yes   
7036                  Yes                  Yes                  Yes   
7037  No internet service  No internet service  No internet service   
7042                  Yes                  Yes                  Yes   

          StreamingMovies        Contract PaperlessBilling  \
0                      No  Month-to-month              Yes   
1                      No        One year               No   
2                      No  Month-to-month              Yes   
3                      No        One year               No   
4                      No  Month-to-month              Yes   
...                   ...             ...              ...   
7034                   No  Month-to-month              Yes   
7035                   No  Month-to-month              Yes   
7036                  Yes        One year               No   
7037  No internet service        Two year              Yes   
7042                  Yes        Two year              Yes   

                  PaymentMethod MonthlyCharges  TotalCharges Churn  
0              Electronic check          29.85         29.85    No  
1                  Mailed check          56.95        1889.5    No  
2                  Mailed check          53.85        108.15   Yes  
3     Bank transfer (automatic)          42.30       1840.75    No  
4              Electronic check          70.70        151.65   Yes  
...                         ...            ...           ...   ...  
7034    Credit card (automatic)         102.95       6886.25   Yes  
7035  Bank transfer (automatic)          7

In [29]:
df2=df.loc[(df['SeniorCitizen']== 1) & (df['Dependents']== "No")]
df2

customerID  gender  SeniorCitizen Partner Dependents  tenure  \
20    8779-QRDMV    Male              1      No         No       1   
30    3841-NFECX  Female              1     Yes         No      71   
31    4929-XIHVW    Male              1     Yes         No       2   
34    3413-BMNZE    Male              1      No         No       1   
50    8012-SOUDQ  Female              1      No         No      43   
...          ...     ...            ...     ...        ...     ...   
7023  1035-IPQPU  Female              1     Yes         No      63   
7029  2235-DWLJU  Female              1      No         No       6   
7031  3605-JISKB    Male              1     Yes         No      55   
7032  6894-LFHLY    Male              1      No         No       1   
7041  8361-LTMKD    Male              1     Yes         No       4   

     PhoneService     MultipleLines InternetService OnlineSecurity  ...  \
20             No  No phone service             DSL             No  ...   
30            Yes               Yes     Fiber optic            Yes  ...   
31            Yes                No     Fiber optic             No  ...   
34            Yes                No             DSL             No  ...   
50            Yes               Yes     Fiber optic             No  ...   
...           ...               ...             ...            ...  ...   
7023          Yes               Yes     Fiber optic             No  ...   
7029           No  No phone service             DSL             No  ...   
7031          Yes               Yes             DSL            Yes  ...   
7032          Yes               Yes     Fiber optic             No  ...   
7041          Yes               Yes     Fiber optic             No  ...   

     DeviceProtection TechSupport StreamingTV StreamingMovies        Contract  \
20                Yes          No          No             Yes  Month-to-month   
30                Yes         Yes          No              No        Two year   
31                Yes          No         Yes             Yes  Month-to-month   
34                 No          No          No              No  Month-to-month   
50                 No          No         Yes              No  Month-to-month   
...               ...         ...         ...             ...             ...   
7023              Yes          No         Yes             Yes  Month-to-month   
7029               No          No         Yes             Yes  Month-to-month   
7031               No          No          No              No        One year   
7032               No          No          No              No  Month-to-month   
7041               No          No          No              No  Month-to-month   

     PaperlessBilling              PaymentMethod MonthlyCharges  TotalCharges  \
20                Yes           Electronic check          39.65         39.65   
30                Yes    Credit card (automatic)          96.35       6766.95   
31                Yes    Credit card (automatic)          95.50        181.65   
34                 No  Bank transfer (automatic)          45.25         45.25   
50                Yes           Electronic check          90.25       3838.75   
...               ...                        ...            ...           ...   
7023              Yes           Electronic check         103.50        6479.4   
7029              Yes           Electronic check          44.40        263.05   
7031               No    Credit card (automatic)          60.00        3316.1   
7032              Yes           Electronic check          75.75         75.75   
7041              Yes               Mailed check          74.40         306.6   

     Churn  
20     Yes  
30      No  
31      No  
34      No  
50      No  
...    ...  
7023    No  
7029    No  
7031    No  
7032   Yes  
7041   Yes  

[1051 rows x 21 columns]

### SeniorCitizen and Tenure

In [31]:
df2=df.loc[(df['SeniorCitizen']== 0) & (df['tenure']>= 12)]
df2

customerID  gender  SeniorCitizen Partner Dependents  tenure  \
1     5575-GNVDE    Male              0      No         No      34   
3     7795-CFOCW    Male              0      No         No      45   
6     1452-KIOVK    Male              0      No        Yes      22   
8     7892-POOKP  Female              0     Yes         No      28   
9     6388-TABGU    Male              0      No        Yes      62   
...          ...     ...            ...     ...        ...     ...   
7036  7750-EYXWZ  Female              0      No         No      12   
7037  2569-WGERO  Female              0      No         No      72   
7038  6840-RESVB    Male              0     Yes        Yes      24   
7039  2234-XADUH  Female              0     Yes        Yes      72   
7042  3186-AJIEK    Male              0      No         No      66   

     PhoneService     MultipleLines InternetService       OnlineSecurity  ...  \
1             Yes                No             DSL                  Yes  ...   
3              No  No phone service             DSL                  Yes  ...   
6             Yes               Yes     Fiber optic                   No  ...   
8             Yes               Yes     Fiber optic                   No  ...   
9             Yes                No             DSL                  Yes  ...   
...           ...               ...             ...                  ...  ...   
7036           No  No phone service             DSL                   No  ...   
7037          Yes                No              No  No internet service  ...   
7038          Yes               Yes             DSL                  Yes  ...   
7039          Yes               Yes     Fiber optic                   No  ...   
7042          Yes                No     Fiber optic                  Yes  ...   

         DeviceProtection          TechSupport          StreamingTV  \
1                     Yes                   No                   No   
3                     Yes                  Yes                   No   
6                      No                   No                  Yes   
8                     Yes                  Yes                  Yes   
9                      No                   No                   No   
...                   ...                  ...                  ...   
7036                  Yes                  Yes                  Yes   
7037  No internet service  No internet service  No internet service   
7038                  Yes                  Yes                  Yes   
7039                  Yes                   No                  Yes   
7042                  Yes                  Yes                  Yes   

          StreamingMovies        Contract PaperlessBilling  \
1                      No        One year               No   
3                      No        One year               No   
6                      No  Month-to-month              Yes   
8                     Yes  Month-to-month              Yes   
9                      No        One year               No   
...                   ...             ...              ...   
7036                  Yes        One year               No   
7037  No internet service        Two year              Yes   
7038                  Yes        One year              Yes   
7039                  Yes        One year              Yes   
7042                  Yes        Two year              Yes   

                  PaymentMethod MonthlyCharges  TotalCharges Churn  
1                  Mailed check          56.95        1889.5    No  
3     Bank transfer (automatic)          42.30       1840.75    No  
6       Credit card (automatic)          89.10        1949.4    No  
8              Electronic check         104.80       3046.05   Yes  
9     Bank transfer (automatic)          56.15       3487.95    No  
...                         ...            ...           ...   ...  
7036           Electronic check          60.65         743.3    No  
7037  Bank transfer (automatic)          2

In [32]:
df2=df.loc[(df['SeniorCitizen']== 0) & (df['tenure']<= 12)]
df2

customerID  gender  SeniorCitizen Partner Dependents  tenure  \
0     7590-VHVEG  Female              0     Yes         No       1   
2     3668-QPYBK    Male              0      No         No       2   
4     9237-HQITU  Female              0      No         No       2   
5     9305-CDSKC  Female              0      No         No       8   
7     6713-OKOMC  Female              0      No         No      10   
...          ...     ...            ...     ...        ...     ...   
7021  1699-HPSBG    Male              0      No         No      12   
7026  8775-CEBBJ  Female              0      No         No       9   
7030  0871-OPBXW  Female              0      No         No       2   
7036  7750-EYXWZ  Female              0      No         No      12   
7040  4801-JZAZL  Female              0     Yes        Yes      11   

     PhoneService     MultipleLines InternetService       OnlineSecurity  ...  \
0              No  No phone service             DSL                   No  ...   
2             Yes                No             DSL                  Yes  ...   
4             Yes                No     Fiber optic                   No  ...   
5             Yes               Yes     Fiber optic                   No  ...   
7              No  No phone service             DSL                  Yes  ...   
...           ...               ...             ...                  ...  ...   
7021          Yes                No             DSL                   No  ...   
7026          Yes                No             DSL                   No  ...   
7030          Yes                No              No  No internet service  ...   
7036           No  No phone service             DSL                   No  ...   
7040           No  No phone service             DSL                  Yes  ...   

         DeviceProtection          TechSupport          StreamingTV  \
0                      No                   No                   No   
2                      No                   No                   No   
4                      No                   No                   No   
5                     Yes                   No                  Yes   
7                      No                   No                   No   
...                   ...                  ...                  ...   
7021                   No                  Yes                  Yes   
7026                   No                   No                   No   
7030  No internet service  No internet service  No internet service   
7036                  Yes                  Yes                  Yes   
7040                   No                   No                   No   

          StreamingMovies        Contract PaperlessBilling  \
0                      No  Month-to-month              Yes   
2                      No  Month-to-month              Yes   
4                      No  Month-to-month              Yes   
5                     Yes  Month-to-month              Yes   
7                      No  Month-to-month               No   
...                   ...             ...              ...   
7021                   No        One year              Yes   
7026                   No  Month-to-month              Yes   
7030  No internet service  Month-to-month              Yes   
7036                  Yes        One year               No   
7040                   No  Month-to-month              Yes   

                  PaymentMethod MonthlyCharges  TotalCharges Churn  
0              Electronic check          29.85         29.85    No  
2                  Mailed check          53.85        108.15   Yes  
4              Electronic check          70.70        151.65   Yes  
5              Electronic check          99.65         820.5   Yes  
7                  Mailed check          29.75         301.9    No  
...                         ...            ...           ...   ...  
7021           Electronic check          59.80         727.8   Yes  
7026  Bank transfer (automatic)          4

In [33]:
df2=df.loc[(df['SeniorCitizen']== 1) & (df['tenure']>= 12)]
df2

customerID  gender  SeniorCitizen Partner Dependents  tenure  \
30    3841-NFECX  Female              1     Yes         No      71   
50    8012-SOUDQ  Female              1      No         No      43   
52    6575-SUVOI  Female              1     Yes         No      25   
54    4667-QONEA  Female              1     Yes        Yes      60   
55    1658-BYGOY    Male              1      No         No      18   
...          ...     ...            ...     ...        ...     ...   
6993  6583-QGCSI  Female              1     Yes         No      50   
7007  2274-XUATA    Male              1     Yes         No      72   
7015  0666-UXTJO    Male              1     Yes         No      34   
7023  1035-IPQPU  Female              1     Yes         No      63   
7031  3605-JISKB    Male              1     Yes         No      55   

     PhoneService     MultipleLines InternetService OnlineSecurity  ...  \
30            Yes               Yes     Fiber optic            Yes  ...   
50            Yes               Yes     Fiber optic             No  ...   
52            Yes               Yes             DSL            Yes  ...   
54            Yes                No             DSL            Yes  ...   
55            Yes               Yes     Fiber optic             No  ...   
...           ...               ...             ...            ...  ...   
6993          Yes               Yes     Fiber optic             No  ...   
7007           No  No phone service             DSL            Yes  ...   
7015          Yes                No     Fiber optic             No  ...   
7023          Yes               Yes     Fiber optic             No  ...   
7031          Yes               Yes             DSL            Yes  ...   

     DeviceProtection TechSupport StreamingTV StreamingMovies        Contract  \
30                Yes         Yes          No              No        Two year   
50                 No          No         Yes              No  Month-to-month   
52                 No         Yes         Yes              No  Month-to-month   
54                Yes         Yes          No             Yes        One year   
55                 No          No         Yes             Yes  Month-to-month   
...               ...         ...         ...             ...             ...   
6993               No          No         Yes              No  Month-to-month   
7007              Yes         Yes         Yes             Yes        Two year   
7015              Yes          No         Yes              No  Month-to-month   
7023              Yes          No         Yes             Yes  Month-to-month   
7031               No          No          No              No        One year   

     PaperlessBilling              PaymentMethod MonthlyCharges  TotalCharges  \
30                Yes    Credit card (automatic)          96.35       6766.95   
50                Yes           Electronic check          90.25       3838.75   
52                Yes    Credit card (automatic)          69.50       1752.65   
54                Yes    Credit card (automatic)          74.85       4456.35   
55                Yes           Electronic check          95.45       1752.55   
...               ...                        ...            ...           ...   
6993              Yes           Electronic check          88.05       4367.35   
7007              Yes  Bank transfer (automatic)          63.10       4685.55   
7015              Yes    Credit card (automatic)          85.20       2874.45   
7023              Yes           Electronic check         103.50        6479.4   
7031               No    Credit card (automatic)          60.00        3316.1   

     Churn  
30      No  
50      No  
52      No  
54      No  
55     Yes  
...    ...  
6993   Yes  
7007    No  
7015    No  
7023    No  
7031    No  

[833 rows x 21 columns]

In [34]:
df2=df.loc[(df['SeniorCitizen']== 1) & (df['tenure']<= 12)]
df2

customerID  gender  SeniorCitizen Partner Dependents  tenure  \
20    8779-QRDMV    Male              1      No         No       1   
31    4929-XIHVW    Male              1     Yes         No       2   
34    3413-BMNZE    Male              1      No         No       1   
53    7495-OOKFY  Female              1     Yes         No       8   
91    2424-WVHPL    Male              1      No         No       1   
...          ...     ...            ...     ...        ...     ...   
7008  1980-KXVPM  Female              1      No         No       3   
7010  0723-DRCLG  Female              1     Yes         No       1   
7029  2235-DWLJU  Female              1      No         No       6   
7032  6894-LFHLY    Male              1      No         No       1   
7041  8361-LTMKD    Male              1     Yes         No       4   

     PhoneService     MultipleLines InternetService OnlineSecurity  ...  \
20             No  No phone service             DSL             No  ...   
31            Yes                No     Fiber optic             No  ...   
34            Yes                No             DSL             No  ...   
53            Yes               Yes     Fiber optic             No  ...   
91            Yes                No     Fiber optic             No  ...   
...           ...               ...             ...            ...  ...   
7008          Yes               Yes     Fiber optic             No  ...   
7010          Yes               Yes     Fiber optic             No  ...   
7029           No  No phone service             DSL             No  ...   
7032          Yes               Yes     Fiber optic             No  ...   
7041          Yes               Yes     Fiber optic             No  ...   

     DeviceProtection TechSupport StreamingTV StreamingMovies        Contract  \
20                Yes          No          No             Yes  Month-to-month   
31                Yes          No         Yes             Yes  Month-to-month   
34                 No          No          No              No  Month-to-month   
53                 No          No          No              No  Month-to-month   
91                 No         Yes          No              No  Month-to-month   
...               ...         ...         ...             ...             ...   
7008               No          No          No              No  Month-to-month   
7010               No          No          No              No  Month-to-month   
7029               No          No         Yes             Yes  Month-to-month   
7032               No          No          No              No  Month-to-month   
7041               No          No          No              No  Month-to-month   

     PaperlessBilling              PaymentMethod MonthlyCharges  TotalCharges  \
20                Yes           Electronic check          39.65         39.65   
31                Yes    Credit card (automatic)          95.50        181.65   
34                 No  Bank transfer (automatic)          45.25         45.25   
53                Yes    Credit card (automatic)          80.65         633.3   
91                 No           Electronic check          74.70          74.7   
...               ...                        ...            ...           ...   
7008              Yes    Credit card (automatic)          75.05        256.25   
7010              Yes           Electronic check          74.45         74.45   
7029              Yes           Electronic check          44.40        263.05   
7032              Yes           Electronic check          75.75         75.75   
7041              Yes               Mailed check          74.40         306.6   

     Churn  
20     Yes  
31      No  
34      No  
53     Yes  
91      No  
...    ...  
7008   Yes  
7010   Yes  
7029    No  
7032   Yes  
7041   Yes  

[320 rows x 21 columns]

### SeniorCitizen and PhoneService

In [ ]:
df2=df.loc[(df['SeniorCitizen']== 0) & (df['tenure']>= 12)]
df2